### ISSUE TITLE
Pyboard D: crash when using WiFi with pyb.stop()

### ISSUE THREAD

**Issue by peterhinch** [micropython/micropython#5152]
This is a regression. The following script publishes once (sometimes more) then crashes producing bouts of garbled output every 10s. In this state it does not respond to ctrl-c.
```python
import pyb
import network
from umqtt.simple import MQTTClient
import time

sta_if = network.WLAN()
sta_if.active(True)
sta_if.connect('my_ssid', 'my_password')  # Adapt
while sta_if.status() in (1, 2):
    time.sleep(1)
time.sleep(1)
if sta_if.isconnected():
    print('Connected', sta_if.ifconfig())

rtc = pyb.RTC()
rtc.wakeup(5000)
while True:
    print('Publish')
    c = MQTTClient('umqtt_client', '192.168.0.10')  # Adapt
    c.connect()
    c.publish(b'result', b'Hello')
    c.disconnect()
    print('Published')
    pyb.stop()
```
Sample output (via FTDI adaptor)
```
>>> import rats27
Connected ('192.168.0.33', '255.255.255.0', '192.168.0.1', '208.67.220.220')
Publish
Published
�␀�␀␀��␀��������␀��␀␀��␀�␀���␀�␀␀���␀�␀␀�����␀�����␀␀����␀�␀␀␀�␀␀␀␀���␀��������␀��������␀��␀␀���␀␀���␀␀␀␀␀␀����␀�␀����␀␀�␀␀␀␀␀������␀���␀␀␀␀␀␀␀�����␀␀�����␀��␀␀�␀���␀␀␀␀␀�␀��␀����␀���␀␀␀␀␀␀�␀␀␀␀␀�␀�␀␀␀␀␀␀�␀��␀␀���␀���␀��␀���␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀�␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀�␀␀␀␀␀␀���␀��␀���␀␀␀␀␀␀�␀␀␀␀␀␀␀␀␀␀␀␀�␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀�␀␀␀␀␀␀␀␀␀␀␀�␀␀�␀␀�␀␀�␀␀␀␀���␀��␀���␀␀␀␀␀␀���␀␀␀␀␀�␀␀��␀␀␀���␀␀�␀�␀�␀␀��␀��������␀��␀␀��␀�␀���␀�␀␀���␀�␀␀�����␀�����␀␀����␀�␀␀␀�␀␀␀␀���␀��������␀��������␀��␀␀���␀␀���␀␀␀␀␀␀����␀�␀����␀␀�␀␀␀␀␀������␀���␀␀␀␀␀␀␀�����␀␀�����␀��␀␀�␀���␀␀␀␀␀�␀��␀����␀���␀␀␀␀␀␀�␀␀␀␀␀�␀�␀␀␀␀␀␀�␀��␀␀���␀���␀��␀���␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀�␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀�␀␀␀␀␀␀���␀��␀���␀␀␀␀␀␀�␀␀␀␀␀␀␀␀␀␀␀␀�␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀␀�␀␀␀␀␀␀␀␀␀␀␀�␀␀�␀␀�␀␀�␀␀␀␀���␀��␀���␀␀␀␀␀␀���␀␀␀␀␀�␀␀��␀␀␀���␀␀�␀Wait 5s
```
The one publication was successfully sent to the broker. Test conditions:
```
MicroPython v1.11-219-gdeedd694a-dirty on 2019-09-25; PYBD-SF2W with STM32F722IEK
```

**Comment by dpgeorge** [comment]
Thanks for the report, and sorry for the slow response here.  I can confirm the issue, and for me it doesn't even need the MQTT part, just being connected to the WiFi is enough:
```python
import machine, net
net.wlan_connect() # my wifi-connect helper
while True:
    print('awake')
    machine.lightsleep(1000)
```

**Comment by dpgeorge** [comment]
The issue was that, upon wake from STOP mode, the IRQs were already enabled (due to QSPI flash being put in deepsleep and reconfiguring the MPU, which reenabled IRQs).  And so the WiFi IRQ handler was being executed immediately after waking from STOP mode, before the system had a chance to reconfigure the clocks after waking.  Thus the WiFi driver was executing with a 16MHz sysclk and the SDMMC interface to the WiFi was not clocking correctly (and UART was outputting at a non-standard baudrate, hence the garbled output).

Fixed by 4f2c737b0c05cd138f15703d492ad64d39c3fcfd
